3-With-Intune MAM/IntuneClient/active-directory-xamarin-intune.Android/EnrollmentNotificationReceiver.cs

```diff
@@ -28,16 +30,33 @@ class EnrollmentNotificationReceiver : Java.Lang.Object, IMAMNotificationReceive
         /// </returns>
         public bool OnReceive(IMAMNotification notification)
         {
+            Debug.WriteLine("***Begin EnrollmentNotificationReceiver -> OnReceive");
             if (notification.Type == MAMNotificationType.MamEnrollmentResult)
             {
+                Debug.WriteLine("***Begin EnrollmentNotificationReceiver -> OnReceive (notification.Type == MAMNotificationType.MamEnrollmentResult)");
                 IMAMEnrollmentNotification enrollmentNotification = notification.JavaCast<IMAMEnrollmentNotification>();
                 MAMEnrollmentManagerResult result = enrollmentNotification.EnrollmentResult;
-
-                if (result.Equals(MAMEnrollmentManagerResult.EnrollmentSucceeded))
+                Debug.WriteLine($"***Begin OnReceive-> Enrollment = {result} Code = {result.Code}");
+            }
+            else if (notification.Type == MAMNotificationType.ComplianceStatus)
+            {
+                Debug.WriteLine("***Begin EnrollmentNotificationReceiver -> OnReceive (notification.Type == MAMNotificationType.ComplianceStatus)");
+                IMAMComplianceNotification complianceNotification = notification.JavaCast<IMAMComplianceNotification>();
+                MAMCAComplianceStatus result = complianceNotification.ComplianceStatus;
+                Debug.WriteLine($"***Begin OnReceive-> Compliance = {result} Code = {result.Code}");
+                if (result.Equals(MAMCAComplianceStatus.Compliant))
                 {
+                    Log.Info(GetType().Name, "remediateCompliance succeeded; status = " + result);
+                    Debug.WriteLine("***Begin OnReceive-> Compliant");
                     // this signals that MAM registration is complete and the app can proceed
                     PCAWrapper.MAMRegsiteredEvent.Set();
                 }
+                else
+                {
+                    Log.Info(GetType().Name, "remediateCompliance failed; status = " + result);
```

Comment:
Can this be a variable?

Reply:
Yes. It can be. But this is just demo app.